server.js

```diff
@@ -20,7 +37,42 @@ app.use(bodyParser.json())
 
 // Start defining your routes here
 app.get('/', (req, res) => {
-  res.send('Hello world')
+  res.send('This is a Happy thoughts API.')
+})
+
+app.get('/thoughts', async (req, res) => {
+  const thoughts = await Thought.find().sort({createdAt: 'desc'}).limit(20).exec()
+  res.json(thoughts)
+})
+
+app.post('/thoughts', async (req, res) => {
+  //Retrieve the information sent by the client to our API endpoint
+  const {message} = req.body
+
+  //Use our mpngoose model to create the database entry
+  const thoughts = new Thought({message})
+
+  try {
+    //Success - the thought is saved to the databse
+    const savedThought = await thoughts.save()
+    res.status(201).json(savedThought)
+    //Bad request
+  } catch (err) {
+    res.status(400).json({message: 'Could not save thought', error: err.errors})
+    }
+})
+
+app.post('/thoughts/:thoughtId/like', async (req, res) => {
+  const {thoughtId} = req.params
+
+  try {
+    // Success - the thought will be updated with a heart like
+    const thoughtLiked = await Thought.updateOne({ _id: thoughtId }, { $inc: { hearts: 1 } });
```

Comment:
Great!
For your information, the method updateOne() is doing the job but the response in Postman is a bit cryptic. If you use the method findOneAndUpdate(), you can get the updated document as response.
```suggestion
    const thoughtLiked = await Thought.findOneAndUpdate({ _id: thoughtId }, { $inc: { hearts: 1 }, { new: true } });
```